gtfs/gtfs_utils/gtfs_utils/config.schema.json

```diff
@@ -0,0 +1,145 @@
+{
+  "schema": {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "http://example.com/config.schema.json",
+    "title": "Main configuration object parameter",
+    "description": "The configuration for a gtfs_stats run",
+    "type": "object",
+    "properties": {
+      "files": {
+        "$ref": "#/definitions/files"
+      },
+      "s3": {
+        "$ref": "#/definitions/s3"
+      },
+      "date_range": {
+        "description": "First and last date of the gtfs files to be download from bucket. {Format: *YYYY-MM-DD*}",
+        "type": "array",
+        "items": {
+          "type": "string",
+          "pattern": "^\\d{4}-\\d{2}-\\d{2}$"
+        },
+        "maxLength": 2,
+        "minLength": 2
+      },
+      "use_data_from_today": {
+        "description": "Download only gtfs data from today (overrides date_range). {default: *False*}",
+        "type": "boolean"
+      },
+      "max_gtfs_size_in_mb": {
+        "description": "Limit the maximum size of the downloaded gtfs files (in MB). If not set the limit is only free disk space.",
+        "type": "integer"
+      },
+      "display_download_progress_bar": {
+        "description": "If true displays a proggress bar while downloading. {default: *False*}",
+        "type": "boolean"
+      },
+      "display_size_on_progress_bar": {
+        "description": "If true displays file size on the download status bar. {default: *False*}",
+        "type": "boolean"
+      },
+      "delete_downloaded_gtfs_zip_files": {
+        "description": "If true delete the gtfs files after parsing them. {default: *True*}",
+        "type": "boolean"
+      },
+      "write_filtered_feed": {
+        "description": "If true writes a filtered version of the gtfs for the specific dates. {default: *False*}",
+        "type": "boolean"
+      },
+      "console_verbosity": {
+        "description": "Lowest logging level to be printed to console. {default: *'Error'*}",
+        "type": "string"
+      }
+    },
+    "required": [
+      "date_range",
```

Comment:
Are you sure it should be required? I'm not sure how such cases should be handled.

Reply:
I can set it as only one of `use_data_from_today` or `date_range` would be required. It's a bit ugly with the date_range defult value:
`date_range: Iterable[str] = ('',)`

Reply:
Why is this related to the default `date_range`?

Reply:
I thought you wanted `date_range` to not be required. I explained it in more depth in your other comment.

Reply:
OK got it. I'm resolving this comment and moving the discussion to the other one.